test/lib/jdk/test/lib/cds/CDSTestUtils.java

```diff
@@ -599,10 +602,79 @@ public static File getSourceFile(String name) {
         return new File(dir, name);
     }
 
+    // Check commandline for the last instance of Xshare to see if the process can load
+    // a CDS archive
+    public static boolean isRunningWithArchive(List<String> cmd) {
+        // -Xshare only works for the java executable
+        if (!cmd.get(0).equals(JDKToolFinder.getJDKTool("java")) || cmd.size() < 2) {
+            return false;
+        }
+
+        // -Xshare options are likely at the end of the args list
+        for (int i  = cmd.size() - 1; i >= 0; i--) {
```

Comment:
nit
```suggestion
        for (int i = cmd.size() - 1; i >= 0; i--) {
```

Reply:
The condition can be "i >= 1" since we know cmd.get(0) is the "java" launcher.